File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
    int minPrice=prices[0];
    int maxprof=0;
    for(int i=1;i<pricesSize;i++)
    {
        minPrice = fmin(minPrice, prices[i]);
        maxprof = fmax(maxprof, prices[i] - minPrice);
    }
    return maxprof;
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = prices[0]
        maxprof = 0
        for i in prices:
            minPrice = min(minPrice, i)
            maxprof = max(maxprof, i - minPrice)
        return maxprof